napari/utils/_tests/test_io.py

```diff
@@ -82,9 +82,9 @@ def raise_no_bigtiff(*args, **kwargs):
 
     monkeypatch.setattr(tifffile, 'imwrite', raise_no_bigtiff)
 
-    data = np.random.randint(
-        low=0, high=2**16, size=(128, 4096, 4096), dtype='uint16'
-    )  # 4GB size
+    # create image data. It can be <4GB compressed because we
+    # monkeypatch tifffile.imwrite to raise an error if bigtiff is not set
+    data = np.empty((20, 200, 200), dtype='uint16')
```

Comment:
```suggestion
    # monkeypatch tifffile.imwrite to raise a struct.error if bigtiff is not set
    # this checks the try/except in io.imsave_tiff resulting in bigtiff being set 
    data = np.empty((20, 200, 200), dtype='uint16')
```

As is, I think the test is rather counter intuitive. Maybe this extra comment helps point people to the right code to make it make sense?

Reply:
At least for me the new version (small file) made no sense and I expected it to fail until I followed the chain from `imsave` to `imsave_tiff`

Reply:
I know, but without this patch, the compression already keeps the size of the image below 4GB, so it was not the big tiff. 

So this is part that require good clarification, but I do not feel to be a good person, to validate if it is.

Reply:
I added a line to the comment. 🤷

Reply:
@jni Could you help to check if it is enough?

Reply:
I suggest adding a docstring to the test to reduce cognitive load and make clear what the test's setup and behavior.

```python
    """Test saving a big tiff file with a size of 4GB.
    
    Set up
    1. Create a 4G image data file of random data.
    2. Ensure that the image file is set to bigtiff.
        We check that it is set to bigtiff since compression may reduce the file size below 4GB.

    Test that imsave saves the bigtiff file correctly.
    """
```

Reply:
Ok. Maybe I should provide more details. 

In general, the TIFF file format specification predicts only 4GB space for data. However, it not, not mean that all array of size above 4GB need to be saved using bigtiff extensions. As we use compression when saving tiff, the bigtiff mode is required, only if the data size **after compression** is above 4GB. 

So when the test was developed, it happened that random-generated data could be compressed to fit in 4GB. 

So we added monkeypatching, to always enforce bigtiff mode in this test. 

So with this PR, we create only small data, but, because of monkeypatch, it is saved as bigtiff. 

The bigtiff is an extension of tiff and could be used with even small files. However, there are still some apps that do not support it. Therefore, we decided to use it only for files that do not fit in 4GB space limit. 

We do not want to fight with generation of real data, that will be really slow. And may create memory problems.

Reply:
Let's see if I can rephrase what you said.

    """Test saving a bigtiff file.

    In our application, we use compression when saving tiff. The bigtiff mode is required, only if the
    data size after compression is above 4GB.
    
    Set up
    1. Create a randomly-generated data file to represent an image. 
    2. Monkeypatch the data file to enforce bigtiff mode. This allows us to
       use a smaller data file to simulate a bigtiff while keeping the test run fast.

    Test that imsave saves the file as bigtiff correctly.
    """

Reply:
Thanks @Czaki. I will leave it up to you to add the docstring to the PR. Thanks!